design/src/rfcs/rfc0007_split_release_process.md

```diff
@@ -0,0 +1,143 @@
+can be enumerated as:
+
+1. All Smithy runtime crates must be published together from `smithy-rs`
+2. AWS runtime crates and the SDK must be published together from `aws-sdk-rust`
+3. CI on `smithy-rs` must give confidence that the Smithy runtime crates,
+   AWS runtime crates, and SDK are all at the right quality bar for publish.
+4. CI on the `aws-sdk-rust` repository must give confidence that the AWS SDK and its
+   runtime crates are at the right quality bar for publish. To do this successfully,
+   it must run against the exact versions of the Smithy runtime crates the code was
+   generated against _both before AND after they have been published to crates.io_.
+
+Background: How Publishing Worked Before
+----------------------------------------
+
+The publish process to crates.io relied on copying all the Smithy runtime crates
+into the final `aws-sdk-rust` repository. Overall, the process looked as follows:
+
+1. `smithy-rs` generates a complete `aws-sdk-rust` source bundle at CI time
+2. The releaser copies the generated bundle over to `aws-sdk-rust`
+3. The releaser runs the `publisher fix-manifests` subcommand to correct the
+   `Cargo.toml` files generated by `smithy-rs`
+4. The `aws-sdk-rust` CI performs one last pass on the code to verify it's sound
+5. The releaser runs the `publisher publish` subcommand to push all the crates up to crates.io
+
+Proposed Solution
+-----------------
+
+CI in `smithy-rs` will be revised to generate two separate build artifacts where it generates
+just an SDK artifact previously. Now, it will have two build targets that get executed from CI
+to generate these artifacts:
+
+- `rust-runtime:assemble` - Generates a publish-ready bundle of Smithy runtime crates.
+- `aws:sdk:assemble` - Generates a publish-ready bundle of AWS runtime crates and SDK crates
+  _without_ the Smithy runtime crates. This build artifact will now also include a file that has
+  the commit hash of `smithy-rs` that it was generated from.
+
+The `aws-sdk-rust` repository will have a new `next` branch that has its own set of CI workflows
+and branch protection rules. The releaser will take the `aws:sdk:assemble` artifact and apply it
+directly to this `next` branch as would have previously been done against the `main` branch.
+The `next` branch's CI will behave similar to the previous CI, except that it runs an additional
+step beforehand to ensure the correct Smithy runtime crates are in place. The process as a whole
+looks as follows:
+
+1. Make a shallow clone of `aws-sdk-rust/next`
+2. Retrieve the `smithy-rs` commit hash that was used to generate the SDK from a file
+   that was generated alongside the rest of the build artifacts from `smithy-rs` and
+   copied into `aws-sdk-rust`.
+3. Make a shallow clone of `smithy-rs` at the correct commit hash
+4. Use a script to add a `[patch]` section to all the AWS SDK crates to point to the
+   Smithy runtime crates from the local clone of `smithy-rs`.
```

Comment:
To preempt questions on this: It needs to be a local clone of `smithy-rs` rather than a git dependency since cargo doesn't provide any way to specify the path within a git repository (the package must be locate-able from the repository root).

Reply:
@rcoh proved me wrong here 😅